### PR TITLE
Fix PDF scroll position on Firefox.

### PR DESCRIPTION
**Pull request by ian-r-rose**
@jasongrout and I had thought that the DOM structure changes from #6255 were responsible for a regression in the PDF.js viewer on Firefox where it lost its scroll position upon unhiding. I did some more testing on this, however, and as far as I can tell it was not the DOM changes, but instead was the re-setting of the data URL.

So before we try a ton of user-agent testing and building out a DOM-structure-matrix, I'd like to try this more limited fix. For me, on Ubuntu+Firefox/Chrome, it seems to behave reasonably. I have no idea why this fixes the PDF.js issue, but overall browser PDF behavior is not something that we can control too much.

I'd appreciate if folks with other OS/browser combinations could test this:
1. Does a PDF render correctly?
1. Does the PDF retain its position upon unhiding or moving?
1. Do markdown links to the document with page fragments work? (e.g. `#page=10`).

### CODE REVIEW

**Comment by blink1073** [comment]
On Windows 10: 

Firefox 66:
Show/hide with page fragment -> scrolls to that fragment
Show/hide no fragment -> scrolls to last position

Chrome 74:
Show/hide with fragment -> scrolls to that fragment
Show/hide no fragment -> scrolls to *top*

**Comment by blink1073** [comment]
Forgot: both keep your scroll position when moving the tab.

**Comment by blink1073** [comment]
Without this change, neither one respects the page fragment.

**Comment by blink1073** [comment]
(well, this change and the previous one).

**Comment by ian-r-rose** [comment]
Some testing on this:

Ubuntu + Firefox: works
Ubuntu + Chrome: works
Windows + Firefox: works
Windows + Chrome: does not work, but does not appear to be regressed

**Comment by blink1073** [comment]
Is it possible to throw out the fragment part of the object `onBeforeHide`?  Then we would avoid it forever trying to return to that fragment.

**Comment by ian-r-rose** [comment]
Thanks for the info @blink1073. I can reproduce the same behavior on Windows 10.

**Comment by ian-r-rose** [comment]
> Is it possible to throw out the fragment part of the object `onBeforeHide`? Then we would avoid it forever trying to return to that fragment.

Ooh, this is a good idea. I was just trying to do that after a timeout, which was not working very well :)

**Comment by ian-r-rose** [comment]
Edits with `before-hide` commit:

On Ubuntu 18.04

Firefox 66:
Show/hide with page fragment -> scrolls to last position
Show/hide with no fragment -> scrolls to last position
Move tab -> maintains position

Chrome 64:
Show/hide with page fragment -> scrolls to last position
Show/hide with no fragment -> scrolls to last position
Move tab -> maintains position

File: packages/pdf-extension/src/index.ts [hunk 1]
```diff
@@ -84,6 +90,18 @@ export class RenderedPDF extends Widget implements IRenderMime.IRenderer {
     return;
   }
 
+  /**
+   * Handle a `before-hide` message.
+   */
+  protected onBeforeHide(): void {
+    // Dispose of any URL fragment before hiding the widget
+    // so that it is not rememberd upon show. Only Firefox
```
Comment (blink1073) [inline packages/pdf-extension/src/index.ts]: remembered*

**Pull request merged**